### PR TITLE
Move pre-commit hooks to use Ruff

### PR DESCRIPTION
**Pull request by patrick91**
**Description**

Ruff is a much faster linter (written in rust), it also supports
sorting imports (replacing isort) and fixing issues.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,14 +5,10 @@ repos:
     hooks:
       - id: black
         args: [--line-length=160]
```
Comment (patrick91) [inline .pre-commit-config.yaml:7]: is 160 really what we want? :)

Comment (patrick91) [reply .pre-commit-config.yaml:7]: ah, btw, the default_stages at the top might not be what we'd like, @sansyrox was there a particular reason it is only set to push?

Comment (sansyrox) [reply .pre-commit-config.yaml:7]: @patrick91 

> is 160 really what we want? :)

This was just something random. I just feel that 70 characters is a bit too less :p

> ah, btw, the default_stages at the top might not be what we'd like, @sansyrox was there a particular reason it is only set to push?

@patrick91 , I am sorry but I don't understand what you mean by this. Can you please explain a bit more? 😅

**Comment by sansyrox** [comment]
Thank you @patrick91 😄 . I have a few questions on the PR.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sansyrox) [reply .pre-commit-config.yaml:7]: > ah, btw, the default_stages at the top might not be what we'd like, @sansyrox was there a particular reason it is only set to push?

Now, I understand. It was mistake 🤣

**Pull request merged**